tidy3d/components/grid/mesher.py

```diff
@@ -373,7 +373,7 @@ def structure_steps(
         min_steps = []
         for structure in structures:
             if isinstance(structure, Structure):
-                if isinstance(structure.medium, (PECMedium, Medium2D)):
+                if structure.medium.is_pec or isinstance(structure.medium, Medium2D):
```

Comment:
So this use the PEC meshing scheme even if the anisotropic is not PEC along some components?

Reply:
good catch, I guess it should use the relevant component for meshing. Although, is this a realistic use case? one component PEC, one component not

Reply:
Maybe for some hyperbolic material that is metallic along some component and dielectric along the rest, and one models the metallic part as PEC.